doc/src/base/punctuation.md

```diff
@@ -54,3 +54,5 @@ Extended documentation for mathematical symbols & functions is [here](@ref math-
 | `=`         | single equals sign is [assignment](@ref man-variables)                                      |
 | [`==`](@ref)| double equals sign is value equality comparison                                             |
 | [`===`](@ref) | triple equals sign is programmatically identical equality comparison.                     |
+| [`=>`](@ref Pair) | right arrow using an equals sign defines a [`Pair`](@ref) typically used to populate [dictionaries](@ref Dictionaries) |
+| `->` | right arrow using a hyphen defines an [anonymous function](@ref man-anonymous-funcitons) on a single line         |
```

Comment:
I did not try to re-build the doc to check, but shouldn't this be `[anonymous function](@ref man-anonymous-functions)`, as in [Anonymous Functions](https://docs.julialang.org/en/v1/manual/functions/#man-anonymous-functions-1)

Reply:
Probably. I don't know how the linking works. I tried inspecting the HTML of the documentation website to get the linking names, but I don't know HTML.  Is the title of the section all I need for the link?

Reply:
I'm not an expert myself, but here is what I'd think you can do in order to get the id without having to know about HTML or anything:
- you want to reference something related to anonymous functions, so you first have to find the relevant page & paragraph in the documentation : this should get you [here](https://docs.julialang.org/en/v1/manual/functions/#man-anonymous-functions-1), which is where you'd like your link to point to.
- clicking on the pencil in the top right part of the page gets you to the markdown source of this part of the documentation (you now know how to do this ; it's the same process you followed to make this pull request in the first place). This should get you [here](https://github.com/JuliaLang/julia/blob/master/doc/src/manual/functions.md).
- now you can look for the relevant [paragraph](https://github.com/JuliaLang/julia/blob/master/doc/src/manual/functions.md#anonymous-functionsid-man-anonymous-functions): its id is written right next to the title: this is the reference you want to use with `(@ref)`.

Thanks and congratulations: not everybody goes through this process of helping improve the documentation!